packages/legend-studio/src/stores/editor-state/GraphGenerationState.ts

```diff
@@ -128,14 +133,31 @@ export class GraphGenerationState {
   get supportedFileGenerationConfigurationsForCurrentElement(): GenerationConfigurationDescription[] {
     if (this.editorStore.currentEditorState instanceof ElementEditorState) {
       const currentElement = this.editorStore.currentEditorState.element;
-      if (
-        currentElement instanceof Class ||
-        currentElement instanceof Enumeration
-      ) {
-        return this.fileGenerationConfigurations
-          .slice()
-          .sort((a, b): number => a.label.localeCompare(b.label));
-      }
+      const getExtraFileGenerationScopeFilters = this.editorStore.pluginManager
+        .getStudioPlugins()
+        .flatMap(
+          (plugin) =>
+            (
+              plugin as DSL_LegendStudioPlugin_Extension
+            ).getExtraFileGenerationScopeFilters?.() ?? [],
+        )
+        .concat([
+          (
+            fileGenerationType: string,
+            packageableElement: PackageableElement,
+          ): boolean =>
+            packageableElement instanceof Class ||
+            packageableElement instanceof Enumeration ||
+            packageableElement instanceof ConcreteFunctionDefinition,
+        ]);
+      return this.fileGenerationConfigurations
+        .slice()
+        .sort((a, b): number => a.label.localeCompare(b.label))
+        .filter((generationType) =>
+          getExtraFileGenerationScopeFilters.some((scopeFilter) =>
+            scopeFilter(generationType.key, currentElement),
```

Comment:
NOTE: **keep this note open, I will handle it in another PR, I think we can let it slide in this one**

I think this design is bad as it creates foot-gun for plugin developer. I saw what you did in `ELMorphir_LegendStudioPlugin` and I realize we let plugin developer to do a bit too much here. Instead, what I think we should do here is just to pass in a configuration of shape:

```ts
{
  type: string;
  filter: (element: PackagableElement) => boolean;
}
```

That way, we can detect all the filters for generation `X` only and apply filter on them with `.some(...)`, when there is not scope filter configuration for a particular generation type, use the generic filter which only allows `Class` and `Enumeration`. This helps avoid affecting other generation types if we messed up the logic in one particular generation type.

Also, we should change the way we show the current file generation dropdown, we should show all and grey out the disabled generation types instead.

Reply:
should be addressed in https://github.com/finos/legend-studio/pull/707/commits/c98340804d98d7b638f29546c479f317c65aaa42